core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java

```diff
@@ -885,11 +885,11 @@ private static boolean isContentAlwaysEmpty(HttpMethod method) {
 
     private static boolean maybeWebSocketUpgrade(AsciiString header, CharSequence value) {
         if (HttpHeaderNames.CONNECTION.contentEqualsIgnoreCase(header) &&
-            HttpHeaderValues.UPGRADE.contentEqualsIgnoreCase(value)) {
+            AsciiString.containsIgnoreCase(value, HttpHeaderValues.UPGRADE)) {
             return true;
         }
         return HttpHeaderNames.UPGRADE.contentEqualsIgnoreCase(header) &&
-               HttpHeaderValues.WEBSOCKET.contentEqualsIgnoreCase(value);
+               AsciiString.containsIgnoreCase(value, HttpHeaderValues.WEBSOCKET);
```

Comment:
I don't think either `websocketxxx` or `xxxwebsocket` is a valid header value. How about splitting the value with a comma `,` before checking the equality?

Reply:
Can we also add a test case that makes sure it doesn't pick up headers like `connection: not-upgrade-lol` and `upgrade: madam-websocket`?